### PR TITLE
Early Sync

### PR DESCRIPTION
**Pull request by maximelb**
- Sync/Configs mechanism for DR Rules only. We can expand to all types in the future (only Rules are needed for the Go lc-service).
- Added a JSON and YAML compat layer to help normalize the datatypes, taken from go-essentials.
- Permission check helper

### CODE REVIEW

File: limacharlie/client.go [hunk 1]
```diff
@@ -355,3 +355,37 @@ func (c *Client) whoAmI() (whoAmIJsonResponse, error) {
...
+			isOrgFound = true
+			break
+		}
+	}
+	if !isOrgFound {
+		return false
+	}
+	if permName == "" {
+		return true
+	}
```
Comment (mireaulf) [inline limacharlie/client.go:384]: That condition feels wrong. I would expect that if the `oid` or the `permName` is empty that function would return false.

File: limacharlie/rule.go [hunk 2]
```diff
@@ -147,6 +138,19 @@ func (d CoreDRRule) Equal(dr CoreDRRule) bool {
...
+	if d.Namespace == "" {
+		d.Namespace = "general"
+	}
+	if dr.Namespace == "" {
+		dr.Namespace = "general"
+	}
+	if d.Namespace != dr.Namespace {
+		return false
+	}
+	return true
```
Comment (mireaulf) [inline limacharlie/rule.go:151]:
```suggestion
	return d.Namespace == dr.Namespace
```

File: limacharlie/sync.go [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+
+type OrgSyncOperation struct {
+	ElementType string `json:"type"`
+	ElementName string `json:"name"`
+	IsAdded     bool   `json:"is_added"`
+	IsRemoved   bool   `json:"is_removed"`
+}
+
+func (org Organization) SyncFetch(options SyncOptions) (OrgConfig, error) {
+	// TODO
```
Comment (mireaulf) [inline limacharlie/sync.go:52]: We could return a specific error type for feature not supported yet? Same idea for the TODOs in SyncPush.

File: limacharlie/client.go (hunk 1 above)
Comment (maximelb) [reply limacharlie/client.go:384]: Empty perm allowed for a general org access check. I split it up into another function, just means more duped code.

**Comment by maximelb** [comment]
PTAL

**Review by mireaulf — APPROVED** [review]
LGTM

**Pull request merged**